lib/index.js

```diff
@@ -1,20 +1,40 @@
-import { print } from 'graphql/language/printer'
-import { parse } from 'graphql/language/parser'
-import crypto from 'crypto'
+import { parseQuery, QueryRepository } from './utils'
 
 const noop = () => { }
 
-export default ({ validateFn, skipValidationFn = noop, validationErrorFn = noop }) =>
-  async (req, res, next) => {
-    if (skipValidationFn(req)) return next()
+export default ({ store, skipValidationFn = noop, validationErrorFn = noop }) => {
+  const repository = new QueryRepository(store)
 
-    req.normalizedQuery = print(parse(req.body.query))
-    req.queryHash = crypto.createHash('sha256').update(req.normalizedQuery).digest('base64')
+  return async (req, res, next) => {
+    if (skipValidationFn(req)) return next()
 
-    if (await validateFn(req.queryHash)) {
-      next()
-    } else {
+    const unauthorized = (error = 'Unauthorized query') => {
       validationErrorFn(req)
-      res.status(401).send('Unauthorized query')
+      res.status(401).json({ error })
+    }
+
+    try {
+      let queryObj = {}
+      const queryId = req.body.queryId || req.query.queryId
+
+      if (queryId) {
+        queryObj = { queryId }
+      } else if (req.body.query) {
+        queryObj = parseQuery(req.body.query)
+      }
+
+      const { query, enabled } = (await repository.get(queryObj.queryId)) || {}
+
+      req.queryId = queryObj.queryId
+      req.body.query = req.normalizedQuery = query
+
+      query && enabled ? next() : unauthorized()
```

Comment:
Si se pide un `queryId` que no existe, va a devolver `unauthorized`. No debería devolver un `query id not found`?